### PR TITLE
[Sql Instrumentation] Unify exposed public API

### PR DESCRIPTION
**Pull request by CodeBlanch**
Relates to #3793

## Changes

* Exposes the same API for all targets to fix package resolution issues when consuming Sql instrumentation into netstandard2.0 projects.

## Public API Changes

```diff
namespace OpenTelemetry.Instrumentation.SqlClient
{
    public class SqlClientInstrumentationOptions
    {
-#if NETFRAMEWORK
-       public bool SetDbStatement { get; set; }
-#else
        public bool SetDbStatementForStoredProcedure { get; set; } = true;
        public bool SetDbStatementForText { get; set; }
        public Func<object, bool> Filter { get; set; }
        public bool RecordException { get; set; }
-#endif
        public Action<Activity, string, object> Enrich { get; set; }
        public bool EnableConnectionLevelAttributes { get; set; }
    }
}
```

## TODOs

* [X] Appropriate `CHANGELOG.md` updated for non-trivial changes
* [ ] Changes in public API reviewed

### CODE REVIEW

**Review by vishweshbankwar — APPROVED** [review]
LGTM

File: src/OpenTelemetry.Instrumentation.SqlClient/SqlClientInstrumentationOptions.cs [hunk 1]
```diff
@@ -61,95 +64,121 @@ public class SqlClientInstrumentationOptions
...
+        /// <item>The first parameter passed to the filter function is the raw
+        /// <c>SqlCommand</c> object for the command being executed.</item>
+        /// <item>The return value for the filter function is interpreted as:
+        /// <list type="bullet">
+        /// <item>If filter returns <see langword="true" />, the command is
+        /// collected.</item>
+        /// <item>If filter returns <see langword="false" /> or throws an
+        /// exception the command is NOT collected.</item>
+        /// </list></item>
+        /// </list>
```
Comment (utpilla) [inline src/OpenTelemetry.Instrumentation.SqlClient/SqlClientInstrumentationOptions.cs:166]: Add the remark about "only supported on .NET and .NET Core" for `Filter` as well?

File: src/OpenTelemetry.Instrumentation.SqlClient/.publicApi/net462/PublicAPI.Unshipped.txt [hunk 2]
```diff
@@ -3,8 +3,14 @@ OpenTelemetry.Instrumentation.SqlClient.SqlClientInstrumentationOptions.EnableCo
 OpenTelemetry.Instrumentation.SqlClient.SqlClientInstrumentationOptions.EnableConnectionLevelAttributes.set -> void
 OpenTelemetry.Instrumentation.SqlClient.SqlClientInstrumentationOptions.Enrich.get -> System.Action<System.Diagnostics.Activity, string, object>
 OpenTelemetry.Instrumentation.SqlClient.SqlClientInstrumentationOptions.Enrich.set -> void
-OpenTelemetry.Instrumentation.SqlClient.SqlClientInstrumentationOptions.SetDbStatement.get -> bool
-OpenTelemetry.Instrumentation.SqlClient.SqlClientInstrumentationOptions.SetDbStatement.set -> void
+OpenTelemetry.Instrumentation.SqlClient.SqlClientInstrumentationOptions.Filter.get -> System.Func<object, bool>
```
Comment (utpilla) [inline src/OpenTelemetry.Instrumentation.SqlClient/.publicApi/net462/PublicAPI.Unshipped.txt:6]: Is `Filter` used on .NET framework as well?

File: src/OpenTelemetry.Instrumentation.SqlClient/CHANGELOG.md [hunk 3]
```diff
@@ -2,6 +2,12 @@
 
 ## Unreleased
 
+* **Breaking change**: The same API is now exposed for `net462` and
+  `netstandard2.0` targets. `SetDbStatement` has been removed. Use
+  `SetDbStatementForText` to capture command text and stored procedure names on
+  .NET Framework.
```
Comment (utpilla) [inline src/OpenTelemetry.Instrumentation.SqlClient/CHANGELOG.md:8]: It's probably a good idea to list out the options that would only be supported on .NET and .NET core in the CHANGELOG.

File: src/OpenTelemetry.Instrumentation.SqlClient/SqlClientInstrumentationOptions.cs (hunk 1 above)
Comment (CodeBlanch) [reply src/OpenTelemetry.Instrumentation.SqlClient/SqlClientInstrumentationOptions.cs:166]: It's there, Line 153 😄

File: src/OpenTelemetry.Instrumentation.SqlClient/CHANGELOG.md (hunk 3 above)
Comment (CodeBlanch) [reply src/OpenTelemetry.Instrumentation.SqlClient/CHANGELOG.md]: I added more text to the CHANGELOG & updated the README. Check it out, LMK what you think!

File: src/OpenTelemetry.Instrumentation.SqlClient/SqlClientInstrumentationOptions.cs (hunk 1 above)
Comment (utpilla) [reply src/OpenTelemetry.Instrumentation.SqlClient/SqlClientInstrumentationOptions.cs:166]: Whoops! 😂

File: src/OpenTelemetry.Instrumentation.SqlClient/README.md [hunk 4]
```diff
@@ -78,7 +78,7 @@ be used to configure how the
 [`db.statement`](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/database.md#call-level-attributes)
 attribute is captured upon execution of a query.
 
-#### .NET Core - SetDbStatementForStoredProcedure and SetDbStatementForText
+#### .NET & .NET Core
 
 On .NET Core, two properties are available: `SetDbStatementForStoredProcedure`
```
Comment (utpilla) [inline src/OpenTelemetry.Instrumentation.SqlClient/README.md:83]:
```suggestion
On .NET and .NET Core, two properties are available: `SetDbStatementForStoredProcedure`
```

**Review by utpilla — APPROVED** [review]

File: src/OpenTelemetry.Instrumentation.SqlClient/CHANGELOG.md (hunk 3 above)
Comment (utpilla) [reply src/OpenTelemetry.Instrumentation.SqlClient/CHANGELOG.md]: Looks good!

**Comment by vishweshbankwar** [comment]
@CodeBlanch - Don't we need to keep this information on **Readme**?

`On .NET Framework, unlike .NET Core, the instrumentation capabilities for both
[`Microsoft.Data.SqlClient`](https://www.nuget.org/packages/Microsoft.Data.SqlClient/)
and `System.Data.SqlClient` are limited:

* [`Microsoft.Data.SqlClient`](https://www.nuget.org/packages/Microsoft.Data.SqlClient/)
  always exposes both the stored procedure name and the full query text but
  doesn't allow for more granular control to turn either on/off depending on
  `CommandType`.
* `System.Data.SqlClient` only exposes stored procedure names and not the full
  query text.`

**Pull request merged**